server/src/main/java/io/atomix/copycat/server/state/ServerSessionManager.java

```diff
@@ -94,10 +94,14 @@ ServerSessionManager unregisterConnection(Connection connection) {
    * Registers a session.
    */
   ServerSessionContext registerSession(ServerSessionContext session) {
+    ServerSessionContext oldSession = clients.remove(session.client());
+    if (oldSession != null) {
+      sessions.remove(oldSession.id());
+    }
     session.setConnection(connections.get(session.client()));
     sessions.put(session.id(), session);
     clients.put(session.client(), session);
-    return session;
+    return oldSession;
```

Comment:
Maybe update the javadoc to include which session is returned and that this can be null?